test/user.js

```diff
@@ -1,29 +1,242 @@
+            expect(alreadyUsedEmails).to.containSubset(emailListResponse);
+            expect(emailListResponse.length).to.equal(usersPerPage);
+            done();
+          })
+      )
+    );
+  });
+
+  it('should return diferent emails in diferent pages', done => {
+    signUpMultipleUsers(usersPerPage * 2).then(emails =>
+      logInAndReturnToken(emails[0]).then(token =>
+        Promise.all([getUsersEmailsInPage(token, 0), getUsersEmailsInPage(token, 1)]).then(pages => {
+          const firstPageEmails = pages[0];
+          const secondPageEmails = pages[1];
+          expect(firstPageEmails.length).to.equal(usersPerPage);
+          expect(secondPageEmails.length).to.equal(usersPerPage);
+          firstPageEmails.forEach(email => expect(secondPageEmails).to.not.contain(email));
+          secondPageEmails.forEach(email => expect(firstPageEmails).to.not.contain(email));
+          done();
+        })
+      )
+    );
+  });
+
+  it('when registering usersPerPage + 1 users, page 1 should contain 1 element', done => {
+    signUpMultipleUsers(usersPerPage + 1).then(emails =>
+      logInAndReturnToken(emails[0]).then(token =>
+        chai
+          .request(server)
+          .get('/users')
+          .set('token', token)
+          .set('page', 1)
+          .send()
+          .then(userListResponse => {
+            const emailListResponse = userListResponse.body.users.map(u => u.email);
+            expect(alreadyUsedEmails).to.containSubset(emailListResponse);
+            expect(emailListResponse.length).to.equal(1);
+            done();
+          })
+      )
+    );
+  });
+
+  it('should fail because the token is missing', done => {
+    signUpTestUserAndReturnEmail()
+      .then(email => logInAndReturnToken(email))
+      .then(token =>
+        chai
+          .request(server)
+          .get('/users')
+          .send()
+          .catch(e => {
+            should.equal(e.response.body.internal_code, errors.BAD_REQUEST);
+            should.equal(e.status, 400);
+            expect(e.response.body.message.length).to.equal(
+              2,
+              'We should expect two messages indicating the token is required and cant be empty'
+            );
+            expect(e.response.body.message).to.include(validationErrorMsgs.tokenIsRequired);
+            expect(e.response.body.message).to.include(validationErrorMsgs.tokenCantBeEmpty);
+            done();
+          })
+      );
+  });
+
+  it('should fail because the token is invalid', done => {
+    chai
+      .request(server)
+      .get('/users')
+      .set('token', 'some invalid token')
+      .send()
+      .catch(e => {
+        should.equal(e.response.body.internal_code, errors.BAD_REQUEST);
+        should.equal(e.status, 400);
+        expect(e.response.body.message).to.equal(badRequestErrorMessages.invalidToken);
+        done();
+      });
+  });
+
+  it('should fail because page is empty', done => {
```

Comment:
This test is not correct. If the user dont send page or limit, you must use a default values for page and limit